### PR TITLE
Use new mbedtls APIs

### PR DESCRIPTION
**Pull request by mariotaku**
I also have a unit test for this change. If you want to integrate some tests please let me know.

### CODE REVIEW

**Comment by cgutman** [comment]
This new API seems much more complex and inefficient for our usecase than the existing API.

What's the advantage of this over the existing API?

**Comment by mariotaku** [comment]
On Ubuntu 22.04 and up, mbedtls-2.27 (or 2.25?) marked the old function as deprecated, causes tests to fail. And for mbedtls-3.x, the old function was completely removed.

I have looked into the source code and used a much better approach to do encryption, this time it should be as efficient & simple as the old one.

File: src/PlatformCrypto.c [hunk 1]
```diff
@@ -68,7 +70,8 @@ bool PltEncryptMessage(PPLT_CRYPTO_CONTEXT ctx, int algorithm, int flags,
     outLength = *outputDataLength;
 
     if (tag != NULL) {
-        if (mbedtls_cipher_auth_encrypt(&ctx->ctx, iv, ivLength, NULL, 0, inputData, inputDataLength, outputData, &outLength, tag, tagLength) != 0) {
+        if (mbedtls_gcm_crypt_and_tag(ctx->ctx.cipher_ctx, MBEDTLS_GCM_ENCRYPT, inputDataLength, iv, ivLength, NULL, 0,
```
Comment (cgutman) [inline src/PlatformCrypto.c:73]: It looks like these functions are going away soon too, so we may just need to go back to your original approach :(

https://github.com/Mbed-TLS/mbedtls/issues/1674

File: src/PlatformCrypto.h [hunk 2]
```diff
@@ -3,6 +3,12 @@
 #include <stdbool.h>
 
 #ifdef USE_MBEDTLS
+#include <mbedtls/version.h>
+
+#if MBEDTLS_VERSION_MAJOR >= 3
+#define MBEDTLS_ALLOW_PRIVATE_ACCESS
```
Comment (cgutman) [inline src/PlatformCrypto.h:9]: I'd rather not depend on private fields. The [3.0 migration guide](https://github.com/Mbed-TLS/mbedtls/blob/development/docs/3.0-migration-guide.md#most-structure-fields-are-now-private) says the struct layout may change in incompatible ways even across minor versions.

File: src/PlatformCrypto.c (hunk 1 above)
Comment (mariotaku) [reply src/PlatformCrypto.c]: I see. So If I understand correctly, in GFE's payload the tag is BEFORE data so multiple extra `memmove`s are non-avoidable :(

File: src/PlatformCrypto.h (hunk 2 above)
Comment (mariotaku) [reply src/PlatformCrypto.h]: That's a good point. I reverted to the original commit.

File: src/PlatformCrypto.c (hunk 1 above)
Comment (cgutman) [reply src/PlatformCrypto.c]: Yeah, it looks like all uses of GCM have the tag before the ciphertext. At least they're consistent 😆

**Pull request merged**